main.tf

```diff
@@ -21,15 +21,26 @@ resource "azurerm_virtual_wan" "virtual_wan" {
   type                              = var.type
 }
 
-resource "azurerm_virtual_hub" "virtual_hub" {
-  for_each = local.virtual_hubs != null && length(local.virtual_hubs) > 0 ? local.virtual_hubs : {}
-
-  location                               = each.value.location
-  name                                   = each.value.name
-  resource_group_name                    = coalesce(each.value.resource_group, local.resource_group_name)
-  address_prefix                         = each.value.address_prefix
-  hub_routing_preference                 = each.value.hub_routing_preference
-  tags                                   = try(each.value.tags, {})
-  virtual_router_auto_scale_min_capacity = each.value.virtual_router_auto_scale_min_capacity
-  virtual_wan_id                         = azurerm_virtual_wan.virtual_wan.id
+module "virtual_hubs" {
+  source = "./modules/virtualhub"
+
+  virtual_hubs = {
```

Comment:
to loop the virtual hubs in parent module

```tf
for_each = local.virtual_hubs

name = each.value.name
location = each.value.location
```

Reply:
Again, if we have to capture the state using moved block, we need to keep the code same as before. This is the only way terraform supports it.